File: backend/app/schemas/offer.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class OfferCreate(BaseModel):
    amount: Decimal = Field(gt=0)


class OfferResponse(BaseModel):
    id: UUID
    conversation_id: UUID
    listing_id: UUID
    buyer_id: UUID
    seller_id: UUID
    amount: Decimal
    currency: str
    status: str
    responded_at: datetime | None
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(
        from_attributes=True,
    )


class OfferActionResponse(BaseModel):
    status: str
    listing_status: str | None = None
    transaction_id: UUID | None = None

File: frontend/src/types/offer.ts
export type ConversationOffer = {
  id: string;
  conversation_id: string;
  listing_id: string;
  buyer_id: string;
  seller_id: string;

  amount: string;
  currency: string;

  status:
    | "PENDING"
    | "ACCEPTED"
    | "REJECTED"
    | "CANCELLED";

  responded_at: string | null;
  created_at: string;
};